src/tools/std20/views/test/join.test.cpp

```diff
@@ -0,0 +1,183 @@
+// include Catch2
+#include <catch2/catch_test_macros.hpp>
+
+// what we are testing
+#include "tools/std20/views.hpp"
+
+// other includes
+#include <forward_list>
+#include <list>
+#include <vector>
+#include "tools/std20/algorithm.hpp"
+
+// convenience typedefs
+//using namespace njoy::tools;
+namespace std20 = nano;
+
+SCENARIO( "join_view" ) {
+
+  const std::vector< int > equal = { -2, -1, 0, 1, 2 };
+  const std::vector< int > modified = { -2, 0, 2, 4, 6 };
+
+  GIVEN( "a container with forward iterators" ) {
+
+    std::forward_list< std::forward_list< int > > values = { { -2, -1 }, { 0 }, { 1, 2 } };
+
+    WHEN( "when iterators are used" ) {
+
+      auto chunk = values | std20::views::join;
+      using Range = decltype(chunk);
+      using Iterator = nano::iterator_t< Range >;
+
+      THEN( "the all_view satisfies the required concepts" ) {
+
+		    CHECK( std20::ranges::view< Range > );
+		    CHECK( ! std20::ranges::sized_range < Range > );
+        CHECK( std20::ranges::forward_range< Range > );
+        CHECK( ! std20::ranges::bidirectional_range< Range > );
+		    CHECK( ! std20::ranges::random_access_range < Range > );
+		    CHECK( ! std20::ranges::contiguous_range < Range > );
+		    CHECK( std20::ranges::common_range < Range > );
+      }
+
+      THEN( "the all_view range and iterator associated types are correct" ) {
```

Comment:
"all_view" -> "join" 

Likewise for other THENs in scenario

Reply:
same